code/modules/customitems/item_defines.dm

```diff
@@ -276,7 +276,39 @@
 		to_chat(target, "<span class='notice'>You comb your tail with the [src].</span>")
 		used = 1
 
-/obj/item/device/fluff/desolate_baton_kit //DesolateG: Michael steampunk_witch
+/obj/item/device/fluff/desolate_coat_kit //DesolateG: Michael Smith
+	name = "armored jacket converstion kit"
+	desc = "Flaps of dark fabric, probably used to somehow modify some sort of an armored garment. Won't help with protection, though."
+	icon_state = "modkit"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/device/fluff/desolate_coat_kit/afterattack(atom/target, mob/user, proximity)
+	if(!proximity || !ishuman(user) || user.incapacitated())
+		return
+
+	if(istype(target, /obj/item/clothing/suit/armor))
+		to_chat(user, "<span class='notice'>You modify the appearance of [target].</span>")
+		var/obj/item/clothing/suit/armor/jacket = target
+		jacket.icon_state = "desolate_coat_open"
+		jacket.icon = 'icons/obj/custom_items.dmi'
+		jacket.ignore_suitadjust = 0
+		jacket.suit_adjusted = 1
+		var/has_action = FALSE
+		for(var/datum/action/A in jacket.actions)
+			if(istype(A, /datum/action/item_action/openclose))
+				has_action = TRUE
+		if(!has_action)
+			new /datum/action/item_action/openclose(jacket)//this actually works
+		jacket.adjust_flavour = "unbutton"
+		jacket.species_fit = null
+		jacket.sprite_sheets = null
+		user.update_inv_wear_suit()
+		qdel(src)
+		return
+
+	to_chat(user, "<span class='warning'>You can't modify [target]!</span>")
```

Comment:
This would be better formatted as
```dm
if(!istype(target, /obj/item/clothing/suit/armor))
    to_chat(user, "<span class='warning'>You can't modify [target]!</span>")
    return
//other code here
```